### DISCUSSION TITLE
Email templates for user management

### DISCUSSION THREAD

**Discussion by JenySadadia** [kernelci/kernelci-project#275] [General]
We have recently integrated [`fastapi-users`](https://pypi.org/project/fastapi-users/) package for user management functionalities in the new API. With the integration, we have enabled support for email verification and password reset via email for user accounts. Please see https://github.com/kernelci/kernelci-api/pull/377 for the reference.

The API uses jinja2 templates for sending emails to the user for the above-mentioned requests i.e. email verification, and password reset. The initial email templates have been added [here](https://github.com/kernelci/kernelci-api/tree/main/templates). These templates are basic and contain only the required information for the user.

We would love to hear from the community about how we can make these emails better and how they should look from the user's pov.
I'll add the templates in separate comments for further discussion.

**Comment by gctucker** [comment]
FWIW I've found it useful to start a separate comment for each section as then people can reply to each of them, rather than a single-threaded discussion.  That's the main difference with GitHub issues I think.  Alternatively, users can quote the part of the discussion description in their comments to be clear what part they're referring to.

**Reply by JenySadadia** [reply]
I'll break down the description into separate comments. Thanks.

**Comment by JenySadadia** [comment]
- Email verification token: This template is used by the API when a user requests a token for email address verification
```
Hello {{ username }},

The token for verifying your email address for KernelCI API account is:

  {{ token }}

Thanks,
KernelCI SysAdmin
```

**Reply by JenySadadia** [reply]
I think it is worth mentioning token validity in the email. Atm, expiry time has been set to `3600 seconds`.

**Comment by JenySadadia** [comment]
- Successful email verification: This template is used by the API when a user's email verification is successful
```
Hello {{ username }},

Your email address has been verified successfully.

Thanks,
KernelCI SysAdmin
```

**Comment by JenySadadia** [comment]
- Password reset token: This template is used by the API to send a password reset token to the user
```
Hello {{ username }},

The token to reset your password for KernelCI API account is:

  {{ token }}

Thanks,
KernelCI SysAdmin

**Comment by JenySadadia** [comment]
- Successful password reset: This template is used to send a notification of a successful password reset
```
Hello {{ username }},

Your password reset is successful.

Thanks,
KernelCI SysAdmin
```
